cosmetics-web/spec/models/notification_spec.rb

```diff
@@ -193,4 +193,35 @@
       end
     end
   end
+
+  describe "#destroy_notification!" do
+    before { notification }
+    let(:responsible_person) { create(:responsible_person_with_user, :with_a_contact_person) }
+    let(:submit_user) { responsible_person.responsible_person_users.first.user }
+
+    context "when is draft" do
+      let(:notification) { create(:draft_notification, responsible_person: responsible_person) }
+
+      it "uses #destroy!" do
+        expect {
+          notification.destroy_notification!(submit_user)
+        }.to change(described_class, :count).by(-1)
+      end
+    end
+
+    context "when is completed" do
+      let(:notification) { create(:registered_notification, responsible_person: responsible_person) }
+      let(:service) { double }
+
+      it "uses NotificationDeleteService" do
+        allow(NotificationDeleteService).to receive(:new) { service }
+        allow(service).to receive(:call)
+
+        notification.destroy_notification!(submit_user)
+
+        expect(NotificationDeleteService).to have_received(:new).with(notification, submit_user) { service }
+        expect(service).to have_received(:call)
+      end
```

Comment:
Suggestion (not a request, I'm fine with this too): 
You can save a couple of lines by stubbing the `call method` in the double object definition and by restricting the return of the stub to when `.new` is called with the proper args. 
In this way the `expect(service).to have_received(:call)` will be enough to test both assertions.
```suggestion
      let(:service) { instance_double(NotificationDeleteService, call: nil) }

      it "uses NotificationDeleteService" do
        allow(NotificationDeleteService).to receive(:new).with(notification, submit_user) { service }

        notification.destroy_notification!(submit_user)

        expect(service).to have_received(:call)
      end
```